app/controllers/purchases_controller.rb

```diff
@@ -0,0 +1,69 @@
+class PurchasesController < ApplicationController
+  before_action :set_purchase, only: %i[show edit update destroy]
+
```

Comment:
- [ ] It will be great to make sure that any requests that are sent to your controller the user is authenticated. What is happening at the moment is that if a user who is not logged in tries to access your controller, the app breaks. This is not a very good approach. So in every controller, you need to add this `before_action :authenticate_user!`